python/ray/dataframe/index_metadata.py

```diff
@@ -52,7 +52,7 @@ def _set_index(self, new_index):
     index = property(_get_index, _set_index)
 
     def coords_of(self, key):
-        raise NotImplementedError()
+        self._coord_df.loc[key]
```

Comment:
You shouldn't need to change this in `_IndexMetadataBase`, it should be implemented in `_IndexMetadata`. Either way, `_IndexMetadataBase` is about to be deprecated by #1965